5.0rc1 expected warnings intercepted by sunpy logger

### Description
With Sunpy 3.0.2 or 3.1.0 installed along with Astropy 5.0rc1 , some tests in `cosmology` and `test_logger` are failing on missing expected warnings that are apparently intercepted by Sunpy's logger mechanism.
### Expected behavior
Warnings should get through  to the `pytest.warns` context.
### Actual behavior
```python
===================================================================================== FAILURES =====================================================================================
_________________________________________________________________________ test_equivalencies_asdf[with_H0] _________________________________________________________________________

tmpdir = local('/private/var/folders/bb/f_z2vsjd21d_p0jpqxzhhwhh00027w/T/pytest-of-derek/pytest-65/test_equivalencies_asdf_with_H0'), equiv = <function with_H0 at 0x10fe41430>

    @pytest.mark.skipif(not HAS_ASDF, reason="requires ASDF")
    @pytest.mark.parametrize("equiv", [cu.with_H0])
    def test_equivalencies_asdf(tmpdir, equiv):
        from asdf.tests import helpers
    
        tree = {"equiv": equiv()}
        with (
            pytest.warns(AstropyDeprecationWarning, match="`with_H0`")
            if equiv.__name__ == "with_H0"
            else contextlib.nullcontext()
        ):
    
>           helpers.assert_roundtrip_tree(tree, tmpdir)
E           Failed: DID NOT WARN. No warnings of type (<class 'astropy.utils.exceptions.AstropyDeprecationWarning'>,) was emitted. The list of emitted warnings is: [].

/opt/lib/python3.8/site-packages/astropy/cosmology/tests/test_units.py:406: Failed
------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
127.0.0.1 - - [05/Nov/2021 23:16:04] "GET /test.asdf HTTP/1.1" 200 -
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
-------------------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------------------
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
```
and for the logger
```python
______________________________________________________________________________ test_log_to_file[None] ______________________________________________________________________________

tmpdir = local('/private/var/folders/bb/f_z2vsjd21d_p0jpqxzhhwhh00027w/T/pytest-of-derek/pytest-64/test_log_to_file_None_0'), level = 'INFO'

    @pytest.mark.parametrize(('level'), [None, 'DEBUG', 'INFO', 'WARN', 'ERROR'])
    def test_log_to_file(tmpdir, level):
    
        local_path = tmpdir.join('test.log')
        log_file = local_path.open('wb')
        log_path = str(local_path.realpath())
        orig_level = log.level
    
        try:
            if level is not None:
                log.setLevel(level)
    
            with log.log_to_file(log_path):
                log.error("Error message")
                log.warning("Warning message")
                log.info("Information message")
                log.debug("Debug message")
    
            log_file.close()
        finally:
            log.setLevel(orig_level)
    
        log_file = local_path.open('rb')
        log_entries = log_file.readlines()
        log_file.close()
    
        if level is None:
            # The log level *should* be set to whatever it was in the config
            level = conf.log_level
    
        # Check list length
        if level == 'DEBUG':
            assert len(log_entries) == 4
        elif level == 'INFO':
            assert len(log_entries) == 3
        elif level == 'WARN':
            assert len(log_entries) == 2
        elif level == 'ERROR':
            assert len(log_entries) == 1
    
        # Check list content
    
>       assert eval(log_entries[0].strip())[-3:] == (
            'astropy.tests.test_logger', 'ERROR', 'Error message')
E         File "<string>", line 1
E           2021-11-05 23:10:45,177, astropy.tests.test_logger, ERROR, Error message
E                   ^
E       SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers

/opt/lib/python3.10/site-packages/astropy/tests/test_logger.py:410: SyntaxError
------------------------------------------------------------------------------- Captured stdout call -------------------------------------------------------------------------------
INFO: Information message [astropy.tests.test_logger]
------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------
ERROR: Error message [astropy.tests.test_logger]
WARNING: Warning message [astropy.tests.test_logger]
-------------------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------------------
ERROR    astropy:test_logger.py:381 Error message
WARNING  astropy:test_logger.py:382 Warning message
INFO     astropy:test_logger.py:383 Information message
```
with corresponding failures for all levels from `DEBUG` to `ERROR`, and
```python
______________________________________________________________________________ test_log_to_file_level ______________________________________________________________________________

tmpdir = local('/private/var/folders/bb/f_z2vsjd21d_p0jpqxzhhwhh00027w/T/pytest-of-derek/pytest-64/test_log_to_file_level0')

    def test_log_to_file_level(tmpdir):
    
        local_path = tmpdir.join('test.log')
        log_file = local_path.open('wb')
        log_path = str(local_path.realpath())
    
        with log.log_to_file(log_path, filter_level='ERROR'):
            log.error("Error message")
            log.warning("Warning message")
    
        log_file.close()
    
        log_file = local_path.open('rb')
        log_entries = log_file.readlines()
        log_file.close()
    
        assert len(log_entries) == 1
>       assert eval(log_entries[0].strip())[-2:] == (
            'ERROR', 'Error message')
E         File "<string>", line 1
E           2021-11-05 23:10:45,282, astropy.tests.test_logger, ERROR, Error message
E                   ^
E       SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers

/opt/lib/python3.10/site-packages/astropy/tests/test_logger.py:443: SyntaxError
------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------
ERROR: Error message [astropy.tests.test_logger]
WARNING: Warning message [astropy.tests.test_logger]
-------------------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------------------
ERROR    astropy:test_logger.py:433 Error message
WARNING  astropy:test_logger.py:434 Warning message
```
and a timeout (I haven't seen this without sunpy installed either, but also do not get it consistently, so might indeed be a connection error):
```python
_________________________________________________________________________________ test_api_lookup __________________________________________________________________________________

    @pytest.mark.remote_data
    def test_api_lookup():
        try:
>           strurl = misc.find_api_page('astropy.utils.misc', 'dev', False,
                                        timeout=5)

/opt/lib/python3.10/site-packages/astropy/utils/tests/test_misc.py:33: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
/opt/lib/python3.10/site-packages/astropy/utils/misc.py:243: in find_api_page
    with get_readable_fileobj(url, encoding='binary', remote_timeout=timeout,
/opt/lib/python3.10/contextlib.py:135: in __enter__
    return next(self.gen)
/opt/lib/python3.10/site-packages/astropy/utils/data.py:267: in get_readable_fileobj
    name_or_obj = download_file(
/opt/lib/python3.10/site-packages/astropy/utils/data.py:1396: in download_file
    raise errors[sources[0]]
/opt/lib/python3.10/site-packages/astropy/utils/data.py:1359: in download_file
    f_name = _download_file_from_source(
/opt/lib/python3.10/site-packages/astropy/utils/data.py:1163: in _download_file_from_source
    with _try_url_open(source_url, timeout=timeout, http_headers=http_headers,
/opt/lib/python3.10/site-packages/astropy/utils/data.py:1100: in _try_url_open
    return urlopener.open(req, timeout=timeout)
/opt/lib/python3.10/urllib/request.py:519: in open
    response = self._open(req, data)
/opt/lib/python3.10/urllib/request.py:536: in _open
    result = self._call_chain(self.handle_open, protocol, protocol +
/opt/lib/python3.10/urllib/request.py:496: in _call_chain
    result = func(*args)
/opt/lib/python3.10/urllib/request.py:1377: in http_open
    return self.do_open(http.client.HTTPConnection, req)
/opt/lib/python3.10/urllib/request.py:1352: in do_open
    r = h.getresponse()
/opt/lib/python3.10/http/client.py:1368: in getresponse
    response.begin()
/opt/lib/python3.10/http/client.py:317: in begin
    version, status, reason = self._read_status()
/opt/lib/python3.10/http/client.py:278: in _read_status
    line = str(self.fp.readline(_MAXLINE + 1), "iso-8859-1")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <socket.SocketIO object at 0x130643700>, b = <memory at 0x137c06e00>

    def readinto(self, b):
        """Read up to len(b) bytes into the writable buffer *b* and return
        the number of bytes read.  If the socket is non-blocking and no bytes
        are available, None is returned.
    
        If *b* is non-empty, a 0 return value indicates that the connection
        was shutdown at the other end.
        """
        self._checkClosed()
        self._checkReadable()
        if self._timeout_occurred:
            raise OSError("cannot read from timed out object")
        while True:
            try:
>               return self._sock.recv_into(b)
E               TimeoutError: timed out

/opt/lib/python3.10/socket.py:705: TimeoutError
```
Notably the `with_H0` warning test in `test_has_expected_equivalencies` succeeds.
Also the logger tests are passing if just called on their own with `package='test'` or 'args='-k test_logger'`.
### Steps to Reproduce
Run `astropy.test()` [all or for `package='cosmology'`] with sunpy installed in the same environment on any of Python 3.8.11, 3.9.6, 3.10.0.

### System Details
Platform: macOS-10.14.6-x86_64-i386-64bit

Executable: /opt/bin/python3.10

Full Python Version: 
3.10.0 (default, Oct  5 2021, 03:51:16) [Clang 11.0.0 (clang-1100.0.33.17)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.21.2
Scipy: 1.7.1
Matplotlib: 3.5.0rc1
h5py: 3.5.0
Pandas: 1.3.3
PyERFA: 2.0.0
Cython: 0.29.24
Scikit-image: not available
asdf: 2.8.1

> I'll note that SunPy being merely installed should have no affect at all on the Astropy test suite. Assuming that @dhomeier forgot to mention that he was importing SunPy, I'd categorize this as a solidly SunPy bug.

Possibly, but this does not depend on importing Sunpy beforehand; I did of course test in a fresh Python session.
Rather it seems to depend also on the `test_equivalencies_asdf` being run  first, as I could not reproduce it when only running the `test_logger` tests. Moreover, running the entire selection a second time, all tests are passing.
I am still seeing the same on 5.0rc2.dev44+gad6566386.
<details><summary>Test Log</summary>

```python
>>> import astropy
>>> astropy.test(args='-vk "test_equivalencies_asdf or test_log_to_file"')
=============================================================================== test session starts ================================================================================
platform darwin -- Python 3.8.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.1 -- /opt/bin/python3.8
cachedir: .pytest_cache

Running tests with Astropy version 5.0rc1.
Running tests in astropy.

Date: 2021-11-08T19:28:23

Platform: macOS-10.14.6-x86_64-i386-64bit

Executable: /opt/bin/python3.8

Full Python Version: 
3.8.12 (default, Nov  6 2021, 01:44:49) 
[Clang 11.0.0 (clang-1100.0.33.17)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.21.2
Scipy: 1.7.1
Matplotlib: 3.4.2
h5py: 2.10.0
Pandas: 1.2.4
PyERFA: 2.0.0
Cython: 0.29.24
Scikit-image: not available
asdf: 2.8.1

Using Astropy options: remote_data: none.

Matplotlib: 3.4.2
Freetype: 2.6.1
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/derek/.hypothesis/examples')
rootdir: /opt/lib/python3.8/site-packages
plugins: arraydiff-0.3, astropy-header-0.1.2, mock-3.6.1, flaky-3.7.0, remotedata-0.3.2, xdist-2.4.0, asdf-2.8.1, forked-1.3.0, openfiles-0.5.0, doctestplus-0.11.0, cov-3.0.0, filter-subpackage-0.1.1, flake8-1.0.7, mpl-0.13, expect-1.1.0, hypothesis-6.8.9
collected 18824 items / 18812 deselected / 3 skipped / 9 selected                                                                                                                  

../../opt/lib/python3.8/site-packages/astropy/cosmology/tests/test_units.py::test_equivalencies_asdf[with_H0] FAILED                                                         [  8%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[None] FAILED                                                                            [ 16%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[DEBUG] FAILED                                                                           [ 25%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[INFO] FAILED                                                                            [ 33%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[WARN] FAILED                                                                            [ 41%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[ERROR] FAILED                                                                           [ 50%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_level FAILED                                                                            [ 58%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_origin1 PASSED                                                                          [ 66%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_origin2 PASSED                                                                          [ 75%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_encoding[] PASSED                                                                       [ 83%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_encoding[utf-8] PASSED                                                                  [ 91%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_encoding[cp1252] PASSED                                                                 [100%]

===================================================================================== FAILURES =====================================================================================
_________________________________________________________________________ test_equivalencies_asdf[with_H0] _________________________________________________________________________

tmpdir = local('/private/var/folders/bb/f_z2vsjd21d_p0jpqxzhhwhh00027w/T/pytest-of-derek/pytest-21/test_equivalencies_asdf_with_H0'), equiv = <function with_H0 at 0x121e01a60>

    @pytest.mark.skipif(not HAS_ASDF, reason="requires ASDF")
    @pytest.mark.parametrize("equiv", [cu.with_H0])
    def test_equivalencies_asdf(tmpdir, equiv):
        from asdf.tests import helpers
    
        tree = {"equiv": equiv()}
        with (
            pytest.warns(AstropyDeprecationWarning, match="`with_H0`")
            if equiv.__name__ == "with_H0"
            else contextlib.nullcontext()
        ):
    
>           helpers.assert_roundtrip_tree(tree, tmpdir)
E           Failed: DID NOT WARN. No warnings of type (<class 'astropy.utils.exceptions.AstropyDeprecationWarning'>,) was emitted. The list of emitted warnings is: [].

/opt/lib/python3.8/site-packages/astropy/cosmology/tests/test_units.py:406: Failed
------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
127.0.0.1 - - [08/Nov/2021 19:28:30] "GET /test.asdf HTTP/1.1" 200 -
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
WARNING: AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead. [astropy.units.equivalencies]
-------------------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------------------
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
WARNING  sunpy:logger.py:211 AstropyDeprecationWarning: `with_H0` is deprecated from `astropy.units.equivalencies` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.with_H0` instead.
______________________________________________________________________________ test_log_to_file[None] ______________________________________________________________________________

tmpdir = local('/private/var/folders/bb/f_z2vsjd21d_p0jpqxzhhwhh00027w/T/pytest-of-derek/pytest-21/test_log_to_file_None_0'), level = 'INFO'

    @pytest.mark.parametrize(('level'), [None, 'DEBUG', 'INFO', 'WARN', 'ERROR'])
    def test_log_to_file(tmpdir, level):
    
        local_path = tmpdir.join('test.log')
        log_file = local_path.open('wb')
        log_path = str(local_path.realpath())
        orig_level = log.level
    
        try:
            if level is not None:
                log.setLevel(level)
    
            with log.log_to_file(log_path):
                log.error("Error message")
                log.warning("Warning message")
                log.info("Information message")
                log.debug("Debug message")
    
            log_file.close()
        finally:
            log.setLevel(orig_level)
    
        log_file = local_path.open('rb')
        log_entries = log_file.readlines()
        log_file.close()
    
        if level is None:
            # The log level *should* be set to whatever it was in the config
            level = conf.log_level
    
        # Check list length
        if level == 'DEBUG':
            assert len(log_entries) == 4
        elif level == 'INFO':
            assert len(log_entries) == 3
        elif level == 'WARN':
            assert len(log_entries) == 2
        elif level == 'ERROR':
            assert len(log_entries) == 1
    
        # Check list content
    
>       assert eval(log_entries[0].strip())[-3:] == (
            'astropy.tests.test_logger', 'ERROR', 'Error message')
E         File "<string>", line 1
E           2021-11-08 19:28:30,197, astropy.tests.test_logger, ERROR, Error message
E                    ^
E       SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers

/opt/lib/python3.8/site-packages/astropy/tests/test_logger.py:410: SyntaxError
------------------------------------------------------------------------------- Captured stdout call -------------------------------------------------------------------------------
INFO: Information message [astropy.tests.test_logger]
------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------
ERROR: Error message [astropy.tests.test_logger]
WARNING: Warning message [astropy.tests.test_logger]
-------------------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------------------
ERROR    astropy:test_logger.py:381 Error message
WARNING  astropy:test_logger.py:382 Warning message
INFO     astropy:test_logger.py:383 Information message
_____________________________________________________________________________ test_log_to_file[DEBUG] ______________________________________________________________________________

tmpdir = local('/private/var/folders/bb/f_z2vsjd21d_p0jpqxzhhwhh00027w/T/pytest-of-derek/pytest-21/test_log_to_file_DEBUG_0'), level = 'DEBUG'

    @pytest.mark.parametrize(('level'), [None, 'DEBUG', 'INFO', 'WARN', 'ERROR'])
    def test_log_to_file(tmpdir, level):
    
        local_path = tmpdir.join('test.log')
        log_file = local_path.open('wb')
        log_path = str(local_path.realpath())
        orig_level = log.level
    
        try:
            if level is not None:
                log.setLevel(level)
    
            with log.log_to_file(log_path):
                log.error("Error message")
                log.warning("Warning message")
                log.info("Information message")
                log.debug("Debug message")
    
            log_file.close()
        finally:
            log.setLevel(orig_level)
    
        log_file = local_path.open('rb')
        log_entries = log_file.readlines()
        log_file.close()
    
        if level is None:
            # The log level *should* be set to whatever it was in the config
            level = conf.log_level
    
        # Check list length
        if level == 'DEBUG':
            assert len(log_entries) == 4
        elif level == 'INFO':
            assert len(log_entries) == 3
        elif level == 'WARN':
            assert len(log_entries) == 2
        elif level == 'ERROR':
            assert len(log_entries) == 1
    
        # Check list content
    
>       assert eval(log_entries[0].strip())[-3:] == (
            'astropy.tests.test_logger', 'ERROR', 'Error message')
E         File "<string>", line 1
E           2021-11-08 19:28:30,220, astropy.tests.test_logger, ERROR, Error message
E                    ^
E       SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers

/opt/lib/python3.8/site-packages/astropy/tests/test_logger.py:410: SyntaxError
------------------------------------------------------------------------------- Captured stdout call -------------------------------------------------------------------------------
INFO: Information message [astropy.tests.test_logger]
DEBUG: Debug message [astropy.tests.test_logger]
------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------
ERROR: Error message [astropy.tests.test_logger]
WARNING: Warning message [astropy.tests.test_logger]
-------------------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------------------
ERROR    astropy:test_logger.py:381 Error message
WARNING  astropy:test_logger.py:382 Warning message
INFO     astropy:test_logger.py:383 Information message
DEBUG    astropy:test_logger.py:384 Debug message
______________________________________________________________________________ test_log_to_file[INFO] ______________________________________________________________________________

tmpdir = local('/private/var/folders/bb/f_z2vsjd21d_p0jpqxzhhwhh00027w/T/pytest-of-derek/pytest-21/test_log_to_file_INFO_0'), level = 'INFO'

    @pytest.mark.parametrize(('level'), [None, 'DEBUG', 'INFO', 'WARN', 'ERROR'])
    def test_log_to_file(tmpdir, level):
    
        local_path = tmpdir.join('test.log')
        log_file = local_path.open('wb')
        log_path = str(local_path.realpath())
        orig_level = log.level
    
        try:
            if level is not None:
                log.setLevel(level)
    
            with log.log_to_file(log_path):
                log.error("Error message")
                log.warning("Warning message")
                log.info("Information message")
                log.debug("Debug message")
    
            log_file.close()
        finally:
            log.setLevel(orig_level)
    
        log_file = local_path.open('rb')
        log_entries = log_file.readlines()
        log_file.close()
    
        if level is None:
            # The log level *should* be set to whatever it was in the config
            level = conf.log_level
    
        # Check list length
        if level == 'DEBUG':
            assert len(log_entries) == 4
        elif level == 'INFO':
            assert len(log_entries) == 3
        elif level == 'WARN':
            assert len(log_entries) == 2
        elif level == 'ERROR':
            assert len(log_entries) == 1
    
        # Check list content
    
>       assert eval(log_entries[0].strip())[-3:] == (
            'astropy.tests.test_logger', 'ERROR', 'Error message')
E         File "<string>", line 1
E           2021-11-08 19:28:30,239, astropy.tests.test_logger, ERROR, Error message
E                    ^
E       SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers

/opt/lib/python3.8/site-packages/astropy/tests/test_logger.py:410: SyntaxError
------------------------------------------------------------------------------- Captured stdout call -------------------------------------------------------------------------------
INFO: Information message [astropy.tests.test_logger]
------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------
ERROR: Error message [astropy.tests.test_logger]
WARNING: Warning message [astropy.tests.test_logger]
-------------------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------------------
ERROR    astropy:test_logger.py:381 Error message
WARNING  astropy:test_logger.py:382 Warning message
INFO     astropy:test_logger.py:383 Information message
______________________________________________________________________________ test_log_to_file[WARN] ______________________________________________________________________________

tmpdir = local('/private/var/folders/bb/f_z2vsjd21d_p0jpqxzhhwhh00027w/T/pytest-of-derek/pytest-21/test_log_to_file_WARN_0'), level = 'WARN'

    @pytest.mark.parametrize(('level'), [None, 'DEBUG', 'INFO', 'WARN', 'ERROR'])
    def test_log_to_file(tmpdir, level):
    
        local_path = tmpdir.join('test.log')
        log_file = local_path.open('wb')
        log_path = str(local_path.realpath())
        orig_level = log.level
    
        try:
            if level is not None:
                log.setLevel(level)
    
            with log.log_to_file(log_path):
                log.error("Error message")
                log.warning("Warning message")
                log.info("Information message")
                log.debug("Debug message")
    
            log_file.close()
        finally:
            log.setLevel(orig_level)
    
        log_file = local_path.open('rb')
        log_entries = log_file.readlines()
        log_file.close()
    
        if level is None:
            # The log level *should* be set to whatever it was in the config
            level = conf.log_level
    
        # Check list length
        if level == 'DEBUG':
            assert len(log_entries) == 4
        elif level == 'INFO':
            assert len(log_entries) == 3
        elif level == 'WARN':
            assert len(log_entries) == 2
        elif level == 'ERROR':
            assert len(log_entries) == 1
    
        # Check list content
    
>       assert eval(log_entries[0].strip())[-3:] == (
            'astropy.tests.test_logger', 'ERROR', 'Error message')
E         File "<string>", line 1
E           2021-11-08 19:28:30,259, astropy.tests.test_logger, ERROR, Error message
E                    ^
E       SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers

/opt/lib/python3.8/site-packages/astropy/tests/test_logger.py:410: SyntaxError
------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------
ERROR: Error message [astropy.tests.test_logger]
WARNING: Warning message [astropy.tests.test_logger]
-------------------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------------------
ERROR    astropy:test_logger.py:381 Error message
WARNING  astropy:test_logger.py:382 Warning message
_____________________________________________________________________________ test_log_to_file[ERROR] ______________________________________________________________________________

tmpdir = local('/private/var/folders/bb/f_z2vsjd21d_p0jpqxzhhwhh00027w/T/pytest-of-derek/pytest-21/test_log_to_file_ERROR_0'), level = 'ERROR'

    @pytest.mark.parametrize(('level'), [None, 'DEBUG', 'INFO', 'WARN', 'ERROR'])
    def test_log_to_file(tmpdir, level):
    
        local_path = tmpdir.join('test.log')
        log_file = local_path.open('wb')
        log_path = str(local_path.realpath())
        orig_level = log.level
    
        try:
            if level is not None:
                log.setLevel(level)
    
            with log.log_to_file(log_path):
                log.error("Error message")
                log.warning("Warning message")
                log.info("Information message")
                log.debug("Debug message")
    
            log_file.close()
        finally:
            log.setLevel(orig_level)
    
        log_file = local_path.open('rb')
        log_entries = log_file.readlines()
        log_file.close()
    
        if level is None:
            # The log level *should* be set to whatever it was in the config
            level = conf.log_level
    
        # Check list length
        if level == 'DEBUG':
            assert len(log_entries) == 4
        elif level == 'INFO':
            assert len(log_entries) == 3
        elif level == 'WARN':
            assert len(log_entries) == 2
        elif level == 'ERROR':
            assert len(log_entries) == 1
    
        # Check list content
    
>       assert eval(log_entries[0].strip())[-3:] == (
            'astropy.tests.test_logger', 'ERROR', 'Error message')
E         File "<string>", line 1
E           2021-11-08 19:28:30,276, astropy.tests.test_logger, ERROR, Error message
E                    ^
E       SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers

/opt/lib/python3.8/site-packages/astropy/tests/test_logger.py:410: SyntaxError
------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------
ERROR: Error message [astropy.tests.test_logger]
-------------------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------------------
ERROR    astropy:test_logger.py:381 Error message
______________________________________________________________________________ test_log_to_file_level ______________________________________________________________________________

tmpdir = local('/private/var/folders/bb/f_z2vsjd21d_p0jpqxzhhwhh00027w/T/pytest-of-derek/pytest-21/test_log_to_file_level0')

    def test_log_to_file_level(tmpdir):
    
        local_path = tmpdir.join('test.log')
        log_file = local_path.open('wb')
        log_path = str(local_path.realpath())
    
        with log.log_to_file(log_path, filter_level='ERROR'):
            log.error("Error message")
            log.warning("Warning message")
    
        log_file.close()
    
        log_file = local_path.open('rb')
        log_entries = log_file.readlines()
        log_file.close()
    
        assert len(log_entries) == 1
>       assert eval(log_entries[0].strip())[-2:] == (
            'ERROR', 'Error message')
E         File "<string>", line 1
E           2021-11-08 19:28:30,291, astropy.tests.test_logger, ERROR, Error message
E                    ^
E       SyntaxError: leading zeros in decimal integer literals are not permitted; use an 0o prefix for octal integers

/opt/lib/python3.8/site-packages/astropy/tests/test_logger.py:443: SyntaxError
------------------------------------------------------------------------------- Captured stderr call -------------------------------------------------------------------------------
ERROR: Error message [astropy.tests.test_logger]
WARNING: Warning message [astropy.tests.test_logger]
-------------------------------------------------------------------------------- Captured log call ---------------------------------------------------------------------------------
ERROR    astropy:test_logger.py:433 Error message
WARNING  astropy:test_logger.py:434 Warning message
================================================================================= warnings summary =================================================================================
../../opt/lib/python3.8/site-packages/_pytest/cacheprovider.py:428
  /opt/lib/python3.8/site-packages/_pytest/cacheprovider.py:428: PytestCacheWarning: could not create cache path /opt/lib/python3.8/site-packages/.pytest_cache/v/cache/nodeids
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../../opt/lib/python3.8/site-packages/_pytest/cacheprovider.py:382
  /opt/lib/python3.8/site-packages/_pytest/cacheprovider.py:382: PytestCacheWarning: could not create cache path /opt/lib/python3.8/site-packages/.pytest_cache/v/cache/lastfailed
    config.cache.set("cache/lastfailed", self.lastfailed)

../../opt/lib/python3.8/site-packages/_pytest/stepwise.py:49
  /opt/lib/python3.8/site-packages/_pytest/stepwise.py:49: PytestCacheWarning: could not create cache path /opt/lib/python3.8/site-packages/.pytest_cache/v/cache/stepwise
    session.config.cache.set(STEPWISE_CACHE_DIR, [])

-- Docs: https://docs.pytest.org/en/stable/warnings.html
============================================================================= short test summary info ==============================================================================
FAILED ../../opt/lib/python3.8/site-packages/astropy/cosmology/tests/test_units.py::test_equivalencies_asdf[with_H0] - Failed: DID NOT WARN. No warnings of type (<class 'astropy...
FAILED ../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[None] -   File "<string>", line 1
FAILED ../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[DEBUG] -   File "<string>", line 1
FAILED ../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[INFO] -   File "<string>", line 1
FAILED ../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[WARN] -   File "<string>", line 1
FAILED ../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[ERROR] -   File "<string>", line 1
FAILED ../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_level -   File "<string>", line 1
======================================================= 7 failed, 5 passed, 3 skipped, 18812 deselected, 3 warnings in 6.91s =======================================================
<ExitCode.TESTS_FAILED: 1>

>>> astropy.test(args='-vk "test_equivalencies_asdf or test_log_to_file"')
=============================================================================== test session starts ================================================================================
platform darwin -- Python 3.8.12, pytest-6.2.5, py-1.10.0, pluggy-0.13.1 -- /opt/bin/python3.8
cachedir: .pytest_cache

Running tests with Astropy version 5.0rc1.
Running tests in astropy.

Date: 2021-11-08T19:31:12

Platform: macOS-10.14.6-x86_64-i386-64bit

Executable: /opt/bin/python3.8

Full Python Version: 
3.8.12 (default, Nov  6 2021, 01:44:49) 
[Clang 11.0.0 (clang-1100.0.33.17)]

encodings: sys: utf-8, locale: UTF-8, filesystem: utf-8
byteorder: little
float info: dig: 15, mant_dig: 15

Package versions: 
Numpy: 1.21.2
Scipy: 1.7.1
Matplotlib: 3.4.2
h5py: 2.10.0
Pandas: 1.2.4
PyERFA: 2.0.0
Cython: 0.29.24
Scikit-image: not available
asdf: 2.8.1

Using Astropy options: remote_data: none.

Matplotlib: 3.4.2
Freetype: 2.6.1
hypothesis profile 'default' -> database=DirectoryBasedExampleDatabase('/Users/derek/.hypothesis/examples')
rootdir: /opt/lib/python3.8/site-packages
plugins: arraydiff-0.3, astropy-header-0.1.2, mock-3.6.1, flaky-3.7.0, remotedata-0.3.2, xdist-2.4.0, asdf-2.8.1, forked-1.3.0, openfiles-0.5.0, doctestplus-0.11.0, cov-3.0.0, filter-subpackage-0.1.1, flake8-1.0.7, mpl-0.13, expect-1.1.0, hypothesis-6.8.9
collected 18828 items / 18816 deselected / 3 skipped / 9 selected                                                                                                                  

../../opt/lib/python3.8/site-packages/astropy/cosmology/tests/test_units.py::test_equivalencies_asdf[with_H0] PASSED                                                         [  8%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[None] PASSED                                                                            [ 16%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[DEBUG] PASSED                                                                           [ 25%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[INFO] PASSED                                                                            [ 33%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[WARN] PASSED                                                                            [ 41%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file[ERROR] PASSED                                                                           [ 50%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_level PASSED                                                                            [ 58%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_origin1 PASSED                                                                          [ 66%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_origin2 PASSED                                                                          [ 75%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_encoding[] PASSED                                                                       [ 83%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_encoding[utf-8] PASSED                                                                  [ 91%]
../../opt/lib/python3.8/site-packages/astropy/tests/test_logger.py::test_log_to_file_encoding[cp1252] PASSED                                                                 [100%]

================================================================================= warnings summary =================================================================================
../../opt/lib/python3.8/site-packages/_pytest/cacheprovider.py:428
  /opt/lib/python3.8/site-packages/_pytest/cacheprovider.py:428: PytestCacheWarning: could not create cache path /opt/lib/python3.8/site-packages/.pytest_cache/v/cache/nodeids
    config.cache.set("cache/nodeids", sorted(self.cached_nodeids))

../../opt/lib/python3.8/site-packages/_pytest/stepwise.py:49
  /opt/lib/python3.8/site-packages/_pytest/stepwise.py:49: PytestCacheWarning: could not create cache path /opt/lib/python3.8/site-packages/.pytest_cache/v/cache/stepwise
    session.config.cache.set(STEPWISE_CACHE_DIR, [])

-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================================================== 12 passed, 3 skipped, 18816 deselected, 2 warnings in 5.79s ============================================================
<ExitCode.OK: 0>
```

</details>

--- a/astropy/cosmology/tests/test_units.py
+++ b/astropy/cosmology/tests/test_units.py
@@ -393,17 +393,11 @@ class Test_with_redshift:
 # they are not in ``astropy.units.equivalencies``, so the following fails
 @pytest.mark.skipif(not HAS_ASDF, reason="requires ASDF")
 @pytest.mark.parametrize("equiv", [cu.with_H0])
-def test_equivalencies_asdf(tmpdir, equiv):
+def test_equivalencies_asdf(tmpdir, equiv, recwarn):
     from asdf.tests import helpers
 
     tree = {"equiv": equiv()}
-    with (
-        pytest.warns(AstropyDeprecationWarning, match="`with_H0`")
-        if equiv.__name__ == "with_H0"
-        else contextlib.nullcontext()
-    ):
-
-        helpers.assert_roundtrip_tree(tree, tmpdir)
+    helpers.assert_roundtrip_tree(tree, tmpdir)
 
 
 def test_equivalency_context_manager():

